triggers/service/src/test-suite/scala/com/daml/lf/engine/trigger/CliSpec.scala

```diff
@@ -0,0 +1,134 @@
+  val confFile = "triggers/service/src/test-suite/resources/trigger-service.conf"
+  def loadCli(file: String): Cli = {
+    Cli.parse(Array("--config", file), Set()).getOrElse(fail("Could not load Cli on parse"))
+  }
+
+  "should pickup the config file provided" in {
+    val file = requiredResource(confFile)
+    val cli = loadCli(file.getAbsolutePath)
+    cli.configFile should not be empty
+  }
+
+  "should be able to successfully load the config based on the file provided" in {
+    val expectedAuthCfg = AuthorizationConfig(
+      authInternalUri = Some(Uri("https://oauth2/internal-uri")),
+      authExternalUri = Some(Uri("https://oauth2/external-uri")),
+      authCallbackUri = Some(Uri("https://oauth2/callback-uri")),
+      authRedirect = AuthClient.RedirectToLogin.Yes,
+    )
+    val expectedJdbcConfig = JdbcConfig(
+      url = "jdbc:postgresql://localhost:5432/test?&ssl=true",
+      driver = "org.postgresql.Driver",
+      user = "postgres",
+      password = "password",
+      poolSize = 12,
+      idleTimeout = 12.seconds,
+      connectionTimeout = 90.seconds,
+      tablePrefix = "foo",
+      minIdle = 4,
+    )
+    val file = requiredResource(confFile)
+    val cli = loadCli(file.getAbsolutePath)
+    cli.configFile should not be empty
+    val cfg = cli.loadFromConfigFile
+    cfg should not be empty
+    cfg.get match {
+      case Left(ex) => fail(s"Failed to load config from config file: ${ex.head.description}")
+      case Right(c) =>
+        c shouldBe minimalConf.copy(
+          darPaths = List(Paths.get("./my-app.dar")),
+          portFile = Some(Paths.get("port-file")),
+          authorization = expectedAuthCfg,
+          triggerStore = Some(expectedJdbcConfig),
+          timeProviderType = TimeProviderType.Static,
+          compilerConfig = Compiler.Config.Dev,
+          initDb = true,
+          ttl = 60.seconds,
+          allowExistingSchema = true,
+        )
+    }
+  }
+
+  "should take default values on loading minimal config" in {
+    val file =
+      requiredResource("triggers/service/src/test-suite/resources/trigger-service-minimal.conf")
+    val cli = loadCli(file.getAbsolutePath)
+    cli.configFile should not be empty
+    val cfg = cli.loadFromConfigFile
+    cfg should not be empty
+    cfg.get match {
+      case Left(ex) => fail(s"Failed to load config from config file: ${ex.head.description}")
+      case Right(c) => c shouldBe minimalConf
+    }
+  }
+
+  "parse should raise error on non-existent config file" in {
+    val cli = loadCli("missingFile.conf")
+    cli.configFile should not be empty
+    val cfg = cli.loadFromConfigFile
+    cfg should not be empty
+    cfg.get match {
+      case Right(_) => fail("Unexpected success trying to load missing config file")
+      case Left(ex) =>
+        inside(ex) { case ConfigReaderFailures(head) =>
+          head shouldBe a[CannotReadFile]
+        }
+    }
```

Comment:
What is good about `inside` is you can get rid of _a lot_ of noise.

```suggestion
    inside(cfg) { case Some(Left(ConfigReaderFailures(head))) =>
      head shouldBe a[CannotReadFile]
    }
```